packages/server/lib/browsers/cri-client.ts

```diff
@@ -172,120 +175,161 @@ type CreateParams = {
   onReconnect?: (client: CriClient) => void
   protocolManager?: ProtocolManagerShape
   fullyManageTabs?: boolean
-  browserClient?: CriClient
+  browserClient?: ICriClient
+  onReconnectAttempt?: (retryIndex: number) => void
 }
 
-export const create = async ({
-  target,
-  onAsynchronousError,
-  host,
-  port,
-  onReconnect,
-  protocolManager,
-  fullyManageTabs,
-  browserClient,
-}: CreateParams): Promise<CriClient> => {
-  const subscriptions: Subscription[] = []
-  const enableCommands: EnableCommand[] = []
-  let enqueuedCommands: EnqueuedCommand[] = []
-
-  let closed = false // has the user called .close on this?
-  let connected = false // is this currently connected to CDP?
-  let crashed = false // has this crashed?
-  let reconnection: Promise<void> | undefined = undefined
-
-  let cri: CDPClient
-  let client: CriClient
-
-  const reconnect = async (retryIndex) => {
-    connected = false
-
-    if (closed) {
-      debug('disconnected, not reconnecting because client is closed %o', { closed, target })
-      enqueuedCommands = []
+export class CriClient implements ICriClient {
+  private subscriptions: Subscription[] = []
+  private enableCommands: EnableCommand[] = []
+  private enqueuedCommands: EnqueuedCommand[] = []
+
+  private _closed = false
+  private _connected = false
+  private crashed = false
+  private reconnection: Promise<void> | undefined = undefined
+
+  private cri: CDPClient | undefined
+
+  constructor (
```

Comment:
Not sure I like converting this to a list of parameters rather than an object. Especially since one of them is a boolean. I tend to like the named parameters better in that case (though I know the tradeoff is the syntactic sugar of declaring them private up front).

Reply:
A fair assessment, but nothing should be directly constructing this class - the static async `create` method is used. I could mark the constructor private, I think, to enforce this.